### DISCUSSION TITLE
How can I get smoother transitions between source changes?

### DISCUSSION THREAD

**Discussion by kingkong-cmd** [openlayers/openlayers#16163] [Q&A] [answered]
I load wind data pngs with `ImageTileSource` and show them with a `WebGLTileLayer` where I draw wind strength by color with an interpolator reading values from the bands of the png.

I have a timeline chooser component where I can change what forecast time to see. It triggers a `source.setUrl(new_url)` where the url contains information about which timestamp i want (`/tile/2024091303/2/1/1`). However, when the source is changed, the transition is first a white flash, then some ms to render the new data (equivalent to whats happening in this example: [https://openlayers.org/en/latest/examples/reusable-source.html](https://openlayers.org/en/latest/examples/reusable-source.html)). 

Is there anything i can do to make this transition smoother? Or are there other source-changing practices I should look into?

**Comment by mike-000** [comment, marked as answer]
You could try a workaround similar to https://stackoverflow.com/a/78128838/10118270

**Reply by kingkong-cmd** [reply]
> You could try a workaround similar to https://stackoverflow.com/a/78128838/10118270

Yes, this looks much better! Only problem now is that the first time I change the layer, the existing layer is not being removed by `map.removeLayer(oldLayer)`. It works in all the subsequent layer changes, meaning I now continuously have two layers on top of each other. Any idea why `removeLayer()` is not playing along?

Edit:
Apologies, it was a typo in the 'loadend' listener where I typed it twice:
`map.once('loadend', () => map.once('loadend', () => map.removeLayer(oldLayer))));`
changed to:
`map.once('loadend', () => map.removeLayer(oldLayer)));
`
```
const oldLayer = map.getLayers().item(0);

const xyz_source = new ImageTileSource({
        url: new_url
})

const image_gl_layer = new WebGLTileLayer({
        source: xyz_source
})

map.once('loadend', () => map.once('loadend', () => map.removeLayer(oldLayer))));
map.getLayers().insertAt(0, image_gl_layer);
```
